### PR TITLE
Inject storage initializer on not prepacked server

### LINKED ISSUE #4251: support use model-initializer with "no-engine"  when modelUri set
For some historical reason, I need serving some custom protocol models(image). I'm not able to [Building a new inference server](https://docs.seldon.io/projects/seldon-core/en/latest/servers/custom.html#building-a-new-inference-server) for these models, so I have to serve them directly with the annotation `seldon.io/no-engine: "true"`. 

Then I cannot use storageInitializer even though the `modelUri` set, the init container not injected.

As an alt, I set the init container in `componentSpecs` by hand, but it's not what I want.

```yaml
apiVersion: machinelearning.seldon.io/v1
kind: SeldonDeployment
metadata:
  name: sklearn-iris
spec:
  predictors:
    - name: default
      replicas: 1
      annotations:
        seldon.io/no-engine: "true"
      graph:
        name: classifier
        type: MODEL
        # modelUri: s3://sklearn/iris
      componentSpecs:
        - spec:
            containers:
              - name: classifier
                image: some-model-image:1.13
                securityContext:
                  privileged: true
                ports:
                  - containerPort: 8080
                    protocol: TCP
                    name: http
                volumeMounts:
                  - mountPath: /mnt/models
                    name: classifier-provision-location
                    readOnly: true
            volumes:
              - name: classifier-provision-location
                emptyDir: {}
            initContainers:
              - name: classifier-model-initializer
                image: seldonio/rclone-storage-initializer:1.15.0-dev
                imagePullPolicy: IfNotPresent
                args:
                  - "s3://sklearn/iris"
                  - "/mnt/models"
                volumeMounts:
                  - mountPath: /mnt/models
                    name: classifier-provision-location
                envFrom:
                  - secretRef:
                      name: seldon-init-container-secret
```

I'm glad to implement this feature if needed

### PR DESCRIPTION
**Pull request by cnfatal**
```release-note
Added Storage Initializers on not prepacked server.
```

**What this PR does / why we need it**:

Enable storage initalizer inject when `modelUri` set in graph on not prepacked server.

**Which issue(s) this PR fixes**: 

Fixes #4251

**Special notes for your reviewer**:

A test already added.

### CODE REVIEW

**Comment by seldondev** [comment]
Hi @cnfatal. Thanks for your PR.

I'm waiting for a [SeldonIO](https://github.com/orgs/SeldonIO/people) or [todo](https://github.com/orgs/todo/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/todo/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository.
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **rafalskolasinski**
You can assign the PR to them by writing `/assign @rafalskolasinski` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: operator/controllers/model_initializer_injector.go
Comment (axsaucedo) [inline operator/controllers/model_initializer_injector.go:73]: Why is this commented?

File: operator/controllers/model_initializer_injector.go
Comment (axsaucedo) [inline operator/controllers/model_initializer_injector.go:84]: Same here

File: operator/controllers/model_initializer_injector.go
Comment (axsaucedo) [inline operator/controllers/model_initializer_injector.go:99]: Why are there fmt changes? Is there a version disparity perhaps?

Comment (axsaucedo) [reply operator/controllers/model_initializer_injector.go:99]: The suggested changes on format look good but best to be added on a separate PR and focus this one on the inject storage initialiser logic

**Review by axsaucedo — CHANGES_REQUESTED** [review]
Thank you for the contribution @cnfatal , I added a couple of comments. It will be important to keep the PR focused so let's make sure only changes are for the feature, and we can keep format chagnes to a separate PR. It will also be important to make sure we have documentation, and we may also need to add an integration test to ensure this feature is properly supported as it's adding a feature in a key path of the codebase.

File: operator/controllers/seldondeployment_controller_test.go
Comment (axsaucedo) [inline operator/controllers/seldondeployment_controller_test.go:1402]: Would be good to test also for normal sdep with (multiple) graph node modelUri

**Comment by RafalSkolasinski** [comment]
Hi @cnfatal, could you post example manifest how you see it being used?

**Comment by cnfatal** [comment]
> Hi @cnfatal, could you post example manifest how you see it being used?

There are some my cases: https://github.com/SeldonIO/seldon-core/issues/4251#issuecomment-1207699328

**Comment by cnfatal** [comment]
Hi @axsaucedo  ,It's My fault, I resolved them in a new commit. Could you please have a look once more.

**Pull request closed without merging**
